contracts/Staking.sol

```diff
@@ -113,148 +97,181 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
     }
 
     /**
-     * @notice Lock `_amount` staked tokens and assign `_manager` as manager with `_data` as data, so they can not be unstaked
+     * @notice Lock `_amount` staked tokens and assign `_lockManager` as manager with `_data` as data, so they can not be unstaked
      * @param _amount The amount of tokens to be locked
-     * @param _manager The manager entity for this particular lock. This entity will have full control over the lock, in particular will be able to unlock it
+     * @param _lockManager The manager entity for this particular lock. This entity will have full control over the lock, in particular will be able to unlock it
      * @param _data Data to parametrize logic for the lock to be enforced by the manager
-     * @return The id of the newly created lock
      */
-    function lock(uint256 _amount, address _manager, bytes _data) external isInitialized returns (uint256) {
+    function lock(uint256 _amount, address _lockManager, uint256 _allowance, bytes _data) external isInitialized {
```

Comment:
I think we need to split this into two methods one to allow the lock manager and another for the lock manager to lock the tokens

Reply:
There’s already `increaseLockAllowance` and `increaseLockAmount`, I can split the first one and create `allowNewLockManager` if that makes it less confusing.